reconcile/terraform_resources.py

```diff
@@ -415,6 +419,15 @@ def run(
         if err:
             cleanup_and_exit(tf, err)
 
+        if defer:
+            defer(
```

Comment:
@maorfr, what do you think?

Reply:
i think you removed `if not light and tf.should_apply:`

Reply:
No, take a look at [line 417](https://github.com/app-sre/qontract-reconcile/blob/288ffab049924f1275455a7989b18e78b4348957/reconcile/terraform_resources.py#L417)